support/packages/local-npm/lib/index.js

```diff
@@ -0,0 +1,190 @@
+/**
+ * SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: MIT
+ */
+
+/* eslint-disable @liferay/no-dynamic-require, no-console */
+
+const childProcess = require('child_process');
+const fs = require('fs');
+const path = require('path');
+
+const LOCAL_REGISTRY = 'http://localhost:4873';
+const NPM_REGISTRY = 'https://registry.npmjs.org/';
+const YARN_REGISTRY = 'https://registry.yarnpkg.com';
+
+module.exports = function main(argv) {
+	switch (argv._[0]) {
+		case 'install':
+		case 'i':
+			install(argv.packagename);
+			break;
+
+		case 'publish':
+		case 'p':
+			publish(argv.projects || ['.']);
+			break;
+
+		case 'registry':
+		case 'r':
+			switch (argv._[1]) {
+				case 'get':
+				case 'g':
+					registryGet();
+					break;
+
+				case 'set':
+				case 's':
+					registrySet(argv._[2]);
+					break;
+
+				default:
+					break;
+			}
+			break;
+
+		default:
+			break;
+	}
+};
+
+function assertLocalRepo() {
+	const npmUrl = run('npm', 'get', 'registry');
+	const yarnUrl = run('yarn', 'config', 'get', 'registry');
+
+	if (npmUrl === NPM_REGISTRY || yarnUrl === YARN_REGISTRY) {
+		console.log('');
+		console.log('  🔴 NOT using LOCAL repo: refusing to do anything');
+		console.log('');
+		process.exit(1);
+	}
+}
+
+function install(packageName) {
+	assertLocalRepo();
+
+	console.log('');
+
+	const dir = path.join('node_modules', packageName);
+
+	console.log(`  ⛔ Removing ${packageName} from node_modules`);
+	if (fs.existsSync(dir)) {
+		fs.rmdirSync(dir, {recursive: true});
+	}
+
+	console.log(`  ⛔ Removing ${packageName} from yarn cache`);
+	run('yarn', 'cache', 'clean', packageName);
+
+	console.log(`  👊 Forcing reinstallation of ${packageName}`);
+	console.log('');
+	run('yarn', 'add', packageName, '--force', '-O', {stdio: 'inherit'});
+
+	console.log('');
+	console.log('  🎉 All work done (exquisitely, we could say)');
+	console.log('');
+}
+
+function publish(projects) {
+	assertLocalRepo();
+
+	projects = projects.map((name) => path.resolve(name));
+
+	console.log('');
```

Comment:
these empty logs in this function seem odd. Can we log a newline or something more explicit? Otherwise I suspect they might get SF removed